### DISCUSSION TITLE
Support for registering buffers which use file backed memory?

### DISCUSSION THREAD

**Discussion by MA-ETL** [axboe/liburing#749] [Q&A]
As I understand, currently io_uring_register() does not currently support registering buffers which use file backed memory. I was wondering, has there been any attempt to enable this functionality and what would be needed in order to make this work?

**Comment by axboe** [comment]
It has some potentially nasty side effects with writeback, which is why it isn't currently allowed.

But I'm curious, what's the motivation for pursuing file backed memory for buffers?

**Reply by axboe** [reply]
Oh, fwiw, should work with file backed huge pages...

**Comment by ollie-etl** [comment]
> But I'm curious, what's the motivation for pursuing file backed memory for buffers?

Working with `mmap`'d driver allocated memory

**Reply by isilence** [reply]
> > But I'm curious, what's the motivation for pursuing file backed memory for buffers?
> 
> Working with `mmap`'d driver allocated memory

We allow some file backed mappings, but they're rather white listed one by one. What driver / type of memory you're specifically interested in?

**Reply by ollie-etl** [reply]
The driver is to a peripheral which produces streams of data via a DMA. The memory is normal DDR4, but is device memory, and the penalty for using system memory is huge (large bandwidth bottleneck). 
This is a Xilinx MPSoC device, to give you an idea of what system is being used / NUMA domains.
